explorer/src/Pos/Explorer/Web/Server.hs

```diff
@@ -186,6 +192,19 @@ explorerHandlers _sendActions =
 -- API Functions
 ----------------------------------------------------------------
 
+getTotalAda :: ExplorerMode ctx m => m CCoin
+getTotalAda = do
+    utxoSum <- EX.getUtxoSum
+    -- getUtxoSum returns Integer, but for external consumers we want to provide Coin.
+    -- However, we don't want to return meaningless values to them, so we add truncation.
+    pure $ mkCCoin $ truncateIntegerToCoin utxoSum
+  where
+    truncateIntegerToCoin :: Integer -> Coin
+    truncateIntegerToCoin n
+        | n < 0 = mkCoin 0
+        | n > coinToInteger (maxBound :: Coin) = maxBound :: Coin
```

Comment:
This is a bit strange to me. We can return an error if the number is larger than the `maxBound`. 
A simple `throwM $ Internal "Coin amount is out of bounds!"` would be more explicit and would alert people that something is wrong.

Reply:
I agree, it seems to be abnormal situation

Reply:
I agree. We shouldn't truncate money info. Throwing exception as @ksaric suggested seems like a right approach here